### PR TITLE
#167996198  Base email sender

### PR DESCRIPTION
**Pull request by Kingobaino**
**What does this PR do?**

  -  Serve as base email sender/service to be reused on the server

  **Description of Task to be completed?**

  - Create an account with sendgrid.com.
  - Generate SEND_GRID_API key.
  - Install @sendgrid/mail on the terminal using npm install --save @sendgrid/mail.
  - Create email.js file in utils folder to handle the email sender
  - Create a function sendEmail 

  **How should this be manually tested?**

  - Open terminal
  - Clone this repo
  - run npm install
  - import 
  - run the following code in your entry file and the to with your own email
const msg = {
      to: 'testing@example.com',
      from: 'cyclopsteam@example.com',
      subject: 'Testing our email sender',
      text: 'Cyclops team is more like a family',
      html: '<strong>Cyclops team is more like a family</strong>',
    };
    sgMail.send(msg)

  **Any background context you want to provide?**

  - None

  **What are the relevant stories?**

  [#167996198](https://www.pivotaltracker.com/story/show/167996198)

  **Screenshots (if appropriate)**

 
Screenshot (9)

  **Questions:**

### CODE REVIEW

File: src/utils/emails.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      text: 'Cyclops team is more like a family',
+      html: '<strong>Cyclops team is more like a family</strong>',
+    };
+    sgMail.send(msg).res.status(201).json({ message: 'email sent successfuly' });
+    next();
+  } catch (error) {
+    return res.status(500).json({ error: 'Email not sent' });
+  };
+};
+export default sendEmail;
```
Comment (dieudonneAwa) [inline src/utils/emails.js:31]: This looks cool. 
But how do I import and use the sendEmail function to send an email message different from the email message you already have in this file? For example, I am working on password reset and some how will need this util to send email to a user. How do I send the mail without having to modify the message you already have in this file? 
Thanks

Comment (Kingobaino) [reply src/utils/emails.js:31]: Well I thought about and i didn't know how to handle it since emails are formatted differently. Any idea is welcome.

**Review by justinefe — COMMENTED** [review]
its very nice maybe you could just add a link or button to it that could actually redirect to an endpoint, and i also think the statuscode should be Ok 200 on succes

**Review by dieudonneAwa — APPROVED** [review]

File: src/utils/emails.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @description function for sending emails to users
+ * @param {String} receiver email of the receipient
+ * @param {String} subject subject of email to be sent
+ * @param {String} content content text to be sent to user
+ * @return {objects} return true for successful email sending or error on failure
+ */
+const sendEmail = async (receiver, subject, content) => {
+  const data = {
+    to: receiver,
+    from: 'cyclopsteam@andela.com',
```
Comment (dieudonneAwa) [inline src/utils/emails.js:20]: Nice work @Kingobaino 
However I think the user of your module should be the one to determine the sender's email

Comment (Wokoro) [reply src/utils/emails.js:20]: Ya, that's true, except there is a dedicated email for that.

**Review by Wokoro — APPROVED** [review]
LGTM

**Review by chukwuemekachm — APPROVED** [review]
LGTM.

**Pull request merged**